### ISSUE TITLE
[4.4.0-beta.2]: Image transforms do not respect 'format' on first call after clearing the 'Asset transform index' cache

### ISSUE THREAD

**Issue by wsydney76** [craftcms/cms#12729]
### Description

Found a lot of dead img src links after updating to 4.4 beta, after refreshing the page, everything was fine.

### Steps to reproduce

1. in `config/general.php` ; `->generateTransformsBeforePageLoad(true)`
2. Clear the  the 'Asset transform index' cache
3. Run this template:

```twig
{% set image = craft.assets.filename('myimage.jpg').one %}
{{ image.url({width: 400, height: 200, format: 'webp'}) }}
<br>
{{ image.url({width: 400, height: 200, format: 'webp'}) }}
```

Output:

```
https://test44.ddev.site/transforms/_400x200_crop_center-center_none/myimage.jpg
https://test44.ddev.site/transforms/_400x200_crop_center-center_none/2/myimage.webp
```

The transform image is generated correctly as `webp`, no `jpg` image is created.

### Expected behavior

Generate correct URL

### Actual behavior

The first call returns an incorrect URL.

### Craft CMS version

4.4.0-beta.2

### PHP version

8.1.13

### Operating system and version

Linux 5.15.79.1-microsoft-standard-WSL2

### Database type and version

MySQL 10.4.27

### Image driver and version

Imagick 3.7.0 (ImageMagick 6.9.11-60)

### Installed plugins and versions

-

**Comment by brandonkelly** [comment]
Thanks for reporting! Fixed for the next 4.4 Beta release, via #12743.

**Comment by brandonkelly** [comment]
4.4.0-beta.3 is out with that fix.
